app/assets/admin/js/settings.js

```diff
@@ -100,56 +118,64 @@ jQuery( document ).ready( function ( $ ) {
-			} else if ( !reg.test( general_videothumb.val() ) ) {
+			} else if ( ! reg.test( general_videothumb.val() ) ) {
 				error_msg += rtmedia_admin_strings.video_thumbnail_invalid_value + ' ' + Math.round( general_videothumb.val() ) + ".";
 				general_videothumb_val = Math.round( general_videothumb.val() );
 			}
-			if ( error_msg != "" ) {
+			if ( '' !== error_msg ) {
 				alert( error_msg );
 				general_videothumb.val( general_videothumb_val );
 				return_code = false;
+
 				return false;
 			}
 		}
 
+		/**
+		 * Validate the media/page general setting.
+		 * Changed the code from showing alert boxes to appending error messages to the selector.
+		 *
+		 * @type {jQuery|HTMLElement|Window.$.fn.init}
+		 */
 		var general_jpeg_image_quality = jQuery( 'input[name^="rtmedia-options[general_jpeg_image_quality]"]' );
-		if ( return_code && general_jpeg_image_quality.length > 0 && typeof general_jpeg_image_quality != "undefined" ) {
-			var error_msg = "";
-			var general_jpeg_image_quality_val = 0;
+		if ( return_code && 1 === general_jpeg_image_quality.length && 'undefined' !== typeof general_jpeg_image_quality ) {
 			if ( general_jpeg_image_quality.val() <= 0 ) {
 				error_msg += rtmedia_admin_strings.jpeg_quality_negative_error;
-				general_jpeg_image_quality_val = 90;
 			} else if ( general_jpeg_image_quality.val() > 100 ) {
 				error_msg += rtmedia_admin_strings.jpeg_quality_percentage_error;
-				general_jpeg_image_quality_val = 100;
-			} else if ( !reg.test( general_jpeg_image_quality.val() ) ) {
+			} else if ( ! reg.test( general_jpeg_image_quality.val() ) ) {
 				error_msg += rtmedia_admin_strings.jpeg_quality_invalid_value + ' ' + Math.round( general_jpeg_image_quality.val() ) + ".";
-				general_jpeg_image_quality_val = Math.round( general_jpeg_image_quality.val() );
 			}
-			if ( error_msg != "" ) {
-				alert( error_msg );
-				general_jpeg_image_quality.val( general_jpeg_image_quality_val );
-				return_code = false;
-				return false;
+			if ( '' !== error_msg ) {
+				general_jpeg_image_quality.next( '.error_msg' ).remove();
+				if( 0 < general_jpeg_image_quality.parents( '.rtm-content' ).attr( 'id' ).length ) {
+					$( '#tab-' + general_jpeg_image_quality.parents( '.rtm-content' ).attr( 'id' ) ).click();
+				}
+
+				return rtp_show_error_message ( general_jpeg_image_quality, error_msg );
 			}
 		}
 
+		/**
+		 * Validate the media/page general setting.
+		 * Changed the code from showing alert boxes to appending error messages to the selector.
+		 *
+		 * @type {jQuery}
+		 */
 		var general_perPageMedia = jQuery( 'input[name^="rtmedia-options[general_perPageMedia]"]' );
-		if ( return_code && general_perPageMedia.length > 0 && typeof general_perPageMedia != "undefined" ) {
-			var error_msg = "";
-			var general_perPageMedia_val = 0;
-			if ( general_perPageMedia.val() < 1 ) {
+		if ( return_code && 1 === general_perPageMedia.length && 'undefined' !== typeof general_perPageMedia ) {
```

Comment:
`'undefined' !== typeof general_perPageMedia` this condition should come before `1 === general_perPageMedia.length`